controllers/cinderapi_controller.go

```diff
@@ -257,22 +257,32 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
 	if instance.Status.APIEndpoints != nil {
 		for _, ksSvc := range keystoneServices {
 
-			ks, err := keystonev1.GetKeystoneServiceWithName(ctx, helper, ksSvc["name"], instance.Namespace)
-
+			// Remove the finalizer from our KeystoneEndpoint CR
+			keystoneEndpoint, err := keystonev1.GetKeystoneEndpointWithName(ctx, helper, ksSvc["name"], instance.Namespace)
 			if err != nil && !k8s_errors.IsNotFound(err) {
 				return ctrl.Result{}, err
 			}
 
-			// Could get here if there was an error and it was "not found", so need to check again
-			// (we do it this way because we want to make sure the "RemoveFinalizer" call is
-			// executed even if the KeystoneService no longer exists)
-			if !k8s_errors.IsNotFound(err) {
-				ksSvcObj := keystonev1.NewKeystoneService(ks.Spec, instance.Namespace, map[string]string{}, 10)
-				err = ksSvcObj.Delete(ctx, helper)
+			if err == nil {
```

Comment:
I see we need to explicitly check for `err` here as the previous condition can't be always met because of the `&&`.
(Same thing applies to the service)